test/test_array_ops.cpp

```diff
@@ -902,6 +902,95 @@ TEST(ArrayOps, Rank) {
   opexecuter.RunTest();
 }  // end of RankOp
 
+// Test op: ScatterNd Op
+TEST(ArrayOps, ScatterNd1D) {
+  Tensor indices(DT_INT32, TensorShape({4, 1}));
+  Tensor updates(DT_FLOAT, TensorShape({4}));
+
+  AssignInputValues<int>(indices, {{2}, {3}, {1}, {7}});
+  AssignInputValues<float>(updates, {9.1, 10.2, -11.3, 12.4});
+
+  vector<int> static_input_indexes = {2};
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  Scope root = Scope::NewRootScope();
+  auto R = ops::ScatterNd(root, indices, updates, {8});
+  std::vector<Output> sess_run_fetchoutputs = {R};
+
+  OpExecuter opexecuter(root, "ScatterNd", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+TEST(ArrayOps, ScatterNdRepeatIndices) {
+  Tensor indices(DT_INT32, TensorShape({4, 1}));
+  Tensor updates(DT_FLOAT, TensorShape({4}));
+
+  // the index "2" appears twice
+  AssignInputValues<int>(indices, {{2}, {3}, {2}, {7}});
+  AssignInputValues<float>(updates, {9.1, 10.2, -11.3, 12.4});
+
+  vector<int> static_input_indexes = {2};
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  Scope root = Scope::NewRootScope();
+  auto R = ops::ScatterNd(root, indices, updates, {10});
+  std::vector<Output> sess_run_fetchoutputs = {R};
+
+  OpExecuter opexecuter(root, "ScatterNd", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
+// This test runs on TF but not on Ng for now
```

Comment:
how are we doing that. From the test it seems it is running on NG

Reply:
Stale comment. Removed it